src/m365/tenant/commands/auditlog/auditlog-report.ts

```diff
@@ -0,0 +1,261 @@
+import * as chalk from 'chalk';
+import auth from '../../../../Auth';
+import { Logger } from '../../../../cli';
+import Command, { CommandOption } from '../../../../Command';
+import GlobalOptions from '../../../../GlobalOptions';
+import request from '../../../../request';
+import Utils from '../../../../Utils';
+import commands from '../../commands';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  contentType: string;
+  startTime?: string;
+  endTime?: string;
+}
+
+interface ActivityfeedSubscription {
+  contentType: string;
+  status: string;
+  webhook: string;
+}
+
+interface AuditContentList {
+  contentType: string;
+  contentId: string;
+  contentUri: string;
+  contentCreated: string;
+  contentExpiration: string;
+}
+
+interface AuditlogReport {
+  CreationTime: string;
+  Id: string;
+  Workload: string;
+  Operation: string;
+  UserId: string;
+}
+
+enum AuditContentTypes {
+  AzureActiveDirectory = "Audit.AzureActiveDirectory",
+  Exchange = "Audit.Exchange",
+  SharePoint = "Audit.SharePoint",
+  General = "Audit.General",
+  DLP = "DLP.All"
+}
+class TenantAuditlogReportCommand extends Command {
+  private serviceUrl: string = 'https://manage.office.com/api/v1.0';
+  private tenantId: string | undefined;
+  private batchedPromises: any = [];
+  private completeAuditReports: AuditlogReport[] = [];
+
+  public get name(): string {
+    return `${commands.TENANT_AUDITLOG_REPORT}`;
+  }
+
+  public get description(): string {
+    return 'Gets audit logs from the Office 365 Management API';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.startTime = args.options.startTime;
+    telemetryProps.endTime = args.options.endTime;
```

Comment:
The starttime and endtime value might not be interesting in our telemetry but if these parameters have been used. You can use this syntax to do it

```
telemetryProps.startTime= typeof args.options.startTime !== 'undefined';
telemetryProps.endTime= typeof args.options.endTime !== 'undefined';
```

On the other hand it would be interesting to see which type of audit is used. Therefore, I would add the option to collect the value for the contenttype

```
telemetryProps.contentType = args.options.contentType;
```

Reply:
Secret between you and me.  _(That is a copy-paste mistake. Promise me you won't tell to anyone.✋)_ 

Will definitely add the `contentType `filter to the telemetry since it will benefit us. Thanks for pointing that out.